NextGenConnect/JSON to HL7v2.xml

```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
           <enabled>true</enabled>
-          <script>// Create url object
+          <script>// WIR message may have 1 PD1 Segment
+
+//PD1-11 - Publicity Code
+
+
+//PD1-12 - Protection Indicator
+
+
+//PD1-13 Effective date for PD1-12
+
+
+//PD1-16 Immunization Registry Status
+
+
+//PD1-17 Effective date for PD1-16
+
+
+//PD1-18 Effective date for PD1-11</script>
+        </com.mirth.connect.plugins.javascriptstep.JavaScriptStep>
+        <com.mirth.connect.plugins.javascriptstep.JavaScriptStep version="3.10.0">
+          <name>NK1 Segment (RE)</name>
+          <sequenceNumber>3</sequenceNumber>
+          <enabled>true</enabled>
+          <script>/*
+// WIR message may have 0+ NK1 Segments
+// Currently deciding to omit this segment
+
+
+// NK1-1 Set ID - Sets the ID for the first related party and so on 1..2..3 for multiple NK1 segments
+// If only listing one contact this always remains 1
+
+tmp[&apos;NK1&apos;][&apos;NK1.1&apos;] = 1;
+
+//NK1-2 Name of responsible person
+
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.1&apos;] = patient[&apos;contact&apos;][1][&apos;family&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.2&apos;] = patient[&apos;contact&apos;][1][&apos;given&apos;][0]; // Stored in array??
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.7&apos;] = &quot;L&quot;;  // Legal Name
+
+//NK1-3 Relationship -- Trick one because of 0063 in WIR and 0131 aren&apos;t compatible in FHIR... use something else in FHIR?
+
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.1&apos;] = patient[&apos;contact&apos;][0][0][1];
+
+// Need a bunch of if statements accordint to page 44 in WIR implementation guide if relationship == &quot;brother&quot; this = BRO
+// if relationship == cargeiver, this = CGV, and so forth
+
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.2&apos;] = &apos;Mother&apos;; // Changes depending on code ^^^^
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.3&apos;] = &apos;HL70063&apos;; // This is the default table... maybe find one compatible with FHIR 0131
+
+//NK1-4 Address of responsible person
+
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.2&apos;] = patient[&apos;contact&apos;][3][&apos;line&apos;][0];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.3&apos;] = patient[&apos;contact&apos;][3][&apos;city&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.4&apos;] = patient[&apos;contact&apos;][3][&apos;state&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.5&apos;] = patient[&apos;contact&apos;][3][&apos;postalCode&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.6&apos;] = patient[&apos;contact&apos;][3][&apos;country&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.7&apos;]= &apos;M&apos; // Mailing address
+
+//NK1-5 Phone number -- need to seperate area code and number
+
+tmp[&apos;NK1&apos;][&apos;NK1.5&apos;][&apos;NK1.5.2&apos;] = patient[&apos;contact&apos;][2][&apos;value&apos;];
```

Comment:
Same as PID, parse it out. Maybe the parsing code could go in a code template or a global script? I haven't explored those too much. Maybe that's how we share code across transformers instead of copy/pasting.

Reply:
I just left this as we are not worrying about NK1